go-controller/pkg/clustermanager/node/subnet_allocator.go

```diff
@@ -359,6 +360,18 @@ func (snr *subnetAllocatorRange) markAllocatedNetwork(owner string, network *net
 
 // allocateNetwork returns a new subnet, or nil if the range is full
 func (snr *subnetAllocatorRange) allocateNetwork(owner string) *net.IPNet {
+	// Return an already allocated subnet instead of creating a new one if a
+	// combination of subnet and node name already exists in the cache
+	for nodeSubnet, nodeName := range snr.allocMap {
+		if nodeName == owner {
+			_, subnet, err := net.ParseCIDR(nodeSubnet)
+			if err != nil {
+				klog.Errorf("Failed to parse subnet %s for node %s: %w", nodeSubnet, nodeName, err)
```

Comment:
I think if you are only printing the error not returning it you can just use %v, why wrap it?